src/routes/plans/[id]/+page.ts

```diff
@@ -15,8 +18,33 @@ export const load: PageLoad = async ({ parent, params, url }) => {
     let initialPlan = await effects.getPlan(planId, user);
 
     if (initialPlan) {
+      // If the plan is in merge review, check if the current user can review the merge.
       if (initialPlan.is_locked) {
-        throw redirect(302, `${base}/plans/${id}/merge`);
+        const initialMergeRequest: PlanMergeRequestSchema | null = await effects.getPlanMergeRequestInProgress(
+          planId,
+          user,
+        );
+
+        if (initialMergeRequest) {
+          const {
+            plan_snapshot_supplying_changes: { plan: sourcePlan },
+            plan_receiving_changes: targetPlan,
+          } = initialMergeRequest;
+
+          const hasReviewPermission = featurePermissions.planBranch.canReviewRequest(
+            user,
+            sourcePlan,
+            targetPlan,
+            initialPlan.model,
+          );
+
+          // If the user can review the merge, take them to the merge review screen.
+          if (hasReviewPermission) {
+            throw redirect(302, `${base}/plans/${id}/merge`);
+          }
+
+          planReadOnly.set(true);
```

Comment:
Thoughts on adding a new store value named `planReadOnlyMergeRequest` that would keep track of the locking status just for merges? It might be worth renaming `planReadOnly` as well to better communicate it's intent.